lib/Doctrine/Query.php

```diff
@@ -1392,7 +1392,11 @@ public function buildSqlQuery($limitSubquery = true)
         $q .= ( ! empty($this->_sqlParts['orderby'])) ? ' ORDER BY ' . implode(', ', $this->_sqlParts['orderby'])  : '';
 
         if ($modifyLimit) {
-            $q = $this->_conn->modifyLimitQuery($q, $this->_sqlParts['limit'], $this->_sqlParts['offset'], false, false, $this);
+            if ($this->_conn instanceof Doctrine_Connection_Mssql) {
```

Comment:
Similar to the other branch above.

`Doctrine_Connection_Mssql` is the only connection object that has this many parameters for the `modifyLimitQuery` method, so only calling the long version if that's the type of connection object.